src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -1879,6 +1895,107 @@ ErrorOnConcurrentRebalance(RebalanceOptions *options)
 }
 
 
+/*
+ * GetColocationId function returns the colocationId of the shard in a PlacementUpdateEvent.
+ */
+static int64
+GetColocationId(PlacementUpdateEvent *move)
+{
+	ShardInterval *shardInterval = LoadShardInterval(move->shardId);
+
+	CitusTableCacheEntry *citusTableCacheEntry = GetCitusTableCacheEntry(
+		shardInterval->relationId);
+
+	return citusTableCacheEntry->colocationId;
+}
+
+
+/*
+ * InitializeMoveDependencyHashMaps function creates the hash maps that we use to track the latest moves so that subsequent moves with the same properties
```

Comment:
style nit: While no strict rule, in general we try to keep comment lines somewhere below 100 characters wide to keep them easy to read in most environments, including the github split-diff view. 

Same note for a few other codecomments.